sdks/python/apache_beam/utils/subprocess_server.py

```diff
@@ -158,9 +158,14 @@ class JavaJarServer(SubprocessServer):
       'local', (threading.local, ),
       dict(__init__=lambda self: setattr(self, 'replacements', {})))()
 
-  def __init__(self, stub_class, path_to_jar, java_arguments):
+  def __init__(self, stub_class, path_to_jar, java_arguments, classpath=None):
+    if classpath:
+      cp_args = ['-classpath', os.pathsep.join(classpath)]
+    else:
+      cp_args = []
     super().__init__(
-        stub_class, ['java', '-jar', path_to_jar] + list(java_arguments))
+        stub_class,
+        ['java'] + cp_args + ['-jar', path_to_jar] + list(java_arguments))
```

Comment:
Unfortunately I don't think "java -jar" command can take a CLASSPATH as an argument. So this probably won't work.
https://docs.oracle.com/en/java/javase/17/docs/specs/man/jar.html

Im fine with partially rolling this back to use a single jar when we automatically startup the expansion service.

Reply:
This is the correct link: https://docs.oracle.com/javase/7/docs/technotes/tools/windows/java.html

"When you use this option, the JAR file is the source of all user classes, and other user class path settings are ignored."

Reply:
Oh, that's really unfortunate. This really limits the ability to use classes in an ad-hoc way... I'll have to think about this some more.

Reply:
OK, I added support for this. I was tempted to just drop the classpath thing, but then the default JavaExternalTransform would be much less useful, and I think this'll be needed for the JDBC transforms as well.